### ISSUE TITLE
How to limit ultralytics yolo11 to use single thread (cpu)

### ISSUE THREAD

**Issue by danielroson** [ultralytics/ultralytics#19071]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

How to limit yolo11 ultralytics to use single thread (CPU) during prediction process.

I have a project using image classifications that I need to limit the use of server resources (threads).

I tried:
results = model(source, device='cpu', workers=1)

or:

export OPENBLAS_NUM_THREADS=1
export NUMEXPR_NUM_THREADS=1
export VECLIB_MAXIMUM_THREADS=1
export NUM_THREADS=1

or:

import torch
torch.set_num_threads(1)
torch.set_num_interop_threads(1)

For the test, the application loads a class that does the inference and another that keeps sending the same image and collecting the result:

```python
    def run(self):

        while True:
            json_result = self.app.classify(passage_image_byte)
            if len(json_result) == 0:
                self.status.inc_image_fail_count()
            else:
                self.status.inc_image_class_count()

```

See below the CPU usage during the test:

![Image](https://github.com/user-attachments/assets/d0e69658-fe24-46ef-a459-f3bfdc8171d2)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @danielroson, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for more information, where many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples are available to assist with various workflows.

If this is a 🐛 Bug Report or unexpected behavior, please share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so that we can better understand and address your issue. This includes the exact code snippet used and the steps to replicate the behavior you are observing.

For custom configuration-related ❓ Questions like this, providing more context about your environment (e.g., OS, version of Python, PyTorch, and ultralytics package) and any specific error logs or messages would help us diagnose the issue more effectively. If you haven't already, ensure you're following [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimal performance.

## Upgrade

First, please ensure you're running the latest version of Ultralytics, as updates often include fixes and optimizations. Upgrade with the following command:

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in a variety of environments. Ensure you're testing in one of our verified setups to see if this resolves the issue:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **AWS** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

Additionally, specifying the `environment.yml` file allows you to recreate our exact dependency setup.

## Status

To verify the integrity of your setup, you can check the status of our CI tests, which run daily to validate functionality against various tasks and modes:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

These tests ensure compatibility on macOS, Windows, and Ubuntu platforms with all features.

We appreciate your contribution to the community by asking this question! 😊 Please note that this is an automated response, and an Ultralytics engineer will follow up with you shortly for further assistance.

**Comment by danielroson** [comment]
Sorry, I found the solution

https://github.com/ultralytics/ultralytics/issues/16724

**Comment by glenn-jocher** [comment]
Great to hear you found a solution! For others seeking to limit YOLOv11 to a single CPU thread, you can set environment variables **before** importing torch/ultralytics:

```python
import os
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["MKL_NUM_THREADS"] = "1"

import torch
torch.set_num_threads(1)

from ultralytics import YOLO
model = YOLO('yolo11n.pt')
results = model(source, device='cpu')
```

This combination of environment variables and thread limits helps restrict CPU parallelism. For more details, refer to [PyTorch CPU parallelism docs](https://pytorch.org/docs/stable/notes/cpu_threading_torchscript_inference.html).
